week3/src/components/main/RandomSelect.jsx

```diff
@@ -0,0 +1,51 @@
+import React, { useEffect, useState } from "react";
+import { menuOptions } from "../../constants/constants";
+import styled, { keyframes } from "styled-components";
+
+const RandomSelect = ({ setFinalRecommendation, setCurrentStep }) => {
+  const [time, setTime] = useState(3);
+
+  useEffect(() => {
+    if (time > 0) {
+      const timer = setTimeout(() => setTime(time - 1), 1000);
+      return () => clearTimeout(timer);
+    } else {
+      selectRandomDish();
+    }
+  }, [time]);
+
+  const selectRandomDish = () => {
+    // 랜덤 음식 추천 로직
+    const cuisines = Object.keys(menuOptions);
+    const randomCuisine = cuisines[Math.floor(Math.random() * cuisines.length)];
+    const types = Object.keys(menuOptions[randomCuisine]);
+    const randomType = types[Math.floor(Math.random() * types.length)];
+    const options = Object.keys(menuOptions[randomCuisine][randomType]);
+    const randomOption = options[Math.floor(Math.random() * options.length)];
+    const randomDish = menuOptions[randomCuisine][randomType][randomOption];
+
+    setFinalRecommendation(`랜덤 추천 음식은 ${randomDish} 입니다.`);
+    setCurrentStep(5);
+  };
+  return (
+    <div>
+      {time ? <CountdownAnimation key={time}>{time}</CountdownAnimation> : null}
```

Comment:
해당 조건이 true일 때만 리턴해야 하는거라면 아래처럼 나타낼 수도 있겠다 !
```suggestion
      {time && <CountdownAnimation key={time}>{time}</CountdownAnimation>}
```

Reply:
아 그러네 && 이거 습관좀 들여야겠따